src/pages/home/report/ReportActionCompose/SuggestionEmoji.tsx

```diff
@@ -1,69 +1,68 @@
-import PropTypes from 'prop-types';
-import React, {useCallback, useEffect, useImperativeHandle, useRef, useState} from 'react';
+import type {ForwardedRef, RefAttributes} from 'react';
+import React, {forwardRef, useCallback, useEffect, useImperativeHandle, useRef, useState} from 'react';
+import type {NativeSyntheticEvent, TextInputSelectionChangeEventData} from 'react-native';
 import {withOnyx} from 'react-native-onyx';
-import _ from 'underscore';
+import type {Emoji} from '@assets/emojis/types';
 import EmojiSuggestions from '@components/EmojiSuggestions';
 import useArrowKeyFocusManager from '@hooks/useArrowKeyFocusManager';
 import useLocalize from '@hooks/useLocalize';
 import * as EmojiUtils from '@libs/EmojiUtils';
 import * as SuggestionsUtils from '@libs/SuggestionUtils';
 import CONST from '@src/CONST';
 import ONYXKEYS from '@src/ONYXKEYS';
-import * as SuggestionProps from './suggestionProps';
+import {isEmptyObject} from '@src/types/utils/EmptyObject';
+import type {SuggestionsRef} from './ReportActionCompose';
+import type {SuggestionProps} from './Suggestions';
+
+type SuggestionsValue = {
+    suggestedEmojis: Emoji[];
+    colonIndex: number;
+    shouldShowSuggestionMenu: boolean;
+};
+
+type SuggestionEmojiOnyxProps = {
+    /** Preferred skin tone */
+    preferredSkinTone: number;
+};
+
+type SuggestionEmojiProps = SuggestionProps &
+    SuggestionEmojiOnyxProps & {
+        /** Function to clear the input */
+        resetKeyboardInput?: () => void;
+    };
 
 /**
  * Check if this piece of string looks like an emoji
- * @param {String} str
- * @param {Number} pos
- * @returns {Boolean}
  */
-const isEmojiCode = (str, pos) => {
+const isEmojiCode = (str: string, pos: number): boolean => {
     const leftWords = str.slice(0, pos).split(CONST.REGEX.SPECIAL_CHAR_OR_EMOJI);
-    const leftWord = _.last(leftWords);
+    const leftWord = leftWords.at(-1) ?? '';
     return CONST.REGEX.HAS_COLON_ONLY_AT_THE_BEGINNING.test(leftWord) && leftWord.length > 2;
 };
 
-const defaultSuggestionsValues = {
+const defaultSuggestionsValues: SuggestionsValue = {
     suggestedEmojis: [],
-    colonSignIndex: -1,
+    colonIndex: -1,
```

Comment:
Please test this change

Reply:
This property was never used 
<img width="409" alt="image" src="https://github.com/Expensify/App/assets/25230417/0ada622a-2b23-4127-a5b8-7d72f92793cf">
even before in the same file they was using `colonIndex` for example here
https://github.com/Expensify/App/blob/d58b4611a30e4c6ed86f3d4b0d690598867cf725/src/pages/home/report/ReportActionCompose/SuggestionEmoji.js#L83-L103